File: year1/semester1/ProgrammingFundamentals/PlanificareVacanta/domain/validator.py
from domain.vacanta import Vacanta
class ValidationException(Exception):
    def __init__(self, message):
        self.message = message

class Validator:
    def validate(self, vacanta: Vacanta):
        errors = []

        if not isinstance(vacanta.id_locatie, int):
            errors.append("ID-ul locatiei trebuie sa fie un numar")

        else:
            if vacanta.id_locatie < 0:
                errors.append("ID-ul locatiei trebuie sa fie un numar natural")

        if vacanta.denumire == "":
            errors.append("Denumirea nu poate fi vida")

        if vacanta.tip == "":
            errors.append("Tip nu poate fi vid")

        if not isinstance(vacanta.pret_pe_zi, int):
            errors.append("Pret pe zi trebuie sa fie un numar")

        else:
            if vacanta.pret_pe_zi < 0:
                errors.append("Pret pe zi trebuie sa fie un numar natural")

        if len(errors) > 0:
            errors_str = '\n'.join(errors)
            raise ValidationException(errors_str)

File: year1/semester2/ObjectOrientedProgramming/Lab8_9_Iteratia2/AdministratorBloc/domain/validator.cpp
#include "validator.h"

void ValidatorLocatar::validate(const Locatar &locatar) const {
    string errors;
    if (locatar.get_apartment() <= 0) {
        errors += "Numarul apartamentului trebuie sa fie natural!\n";
    }
    if (locatar.get_nume().empty()) {
        errors += "Numele nu poate fi vid!\n";
    }
    if (locatar.get_suprafata() <= 0) {
        errors += "Suprafata nu poate fi negativa!\n";
    }
    if (locatar.get_tip().empty()) {
        errors += "Tipul nu poate fi vid\n";
    }

    if (!errors.empty()) {
        throw ValidationException(errors);
    }
}